aviary/interface/reports.py

```diff
@@ -46,6 +51,59 @@ def register_custom_reports():
                     method='run_driver',
                     pre_or_post='post')
 
+    register_report(name='run_status',
+                    func=run_status,
+                    desc='Generates a report on the status of the run',
+                    class_name='AviaryProblem',
+                    method='run_driver',
+                    pre_or_post='post',
+                    )
+
+
+def run_status(prob):
+    """
+    Creates a JSON file that containts high level overview of the run
+
+    Parameters
+    ----------
+    prob : AviaryProblem
+        The AviaryProblem used to generate this report
+    """
+    # Note: Due to a possible bug in OpenMDAO, we need to assign Problem as the
+    # class_name instead of AviaryProblem. Make sure that we don't try to write
+    # aviary reports without aviary in the model.
+
+    from aviary.interface.methods_for_level2 import AviaryProblem
+    if not isinstance(prob, AviaryProblem):
+        return
+
+    reports_folder = Path(prob.get_reports_dir())
+    report_file = reports_folder / 'status.json'
+
+    runtime = prob.driver.result.runtime
+    runtime_ms = (runtime * 1000.0) % 1000.0
+    runtime_formatted = \
+        f"{time.strftime('%H hours %M minutes %S seconds', time.gmtime(runtime))} " \
+        f"{runtime_ms:.1f} milliseconds"
+
+    t = datetime.datetime.now()
+    time_stamp = t.strftime("%Y-%m-%d %H:%M:%S %Z")
+
+    status = {}
+    status['Problem'] = prob._name
+    status['Script'] = sys.argv[0]
+    status['Optimizer'] = prob.driver._get_name()
+    status['Number of driver iterations'] = prob.driver.result.iter_count
+    status['Number of model evals'] = prob.driver.result.model_evals
+    status['Number of deriv evals'] = prob.driver.result.deriv_evals
+    status['Wall clock run time'] = runtime_formatted
+    status['Exit status'] = prob.driver.result.exit_status
+    status['Report generation date and time'] = time_stamp
+
+    with open(report_file, 'w') as f:
```

Comment:
I think you want to make sure to only do this on rank 0 if you are in MPI.

Reply:
Agreed! Fixed